### PR TITLE
 detect: implement http {location,server} sticky buffer

### PR DESCRIPTION
**Pull request by jlucovsky**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/2332) ticket:

Describe changes:
- Add sticky content buffer for http.location to match HTTP response Location:
- Add sticky content buffer for http.server to match HTTP response Server:
- Userguide updates for both sticky buffers

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

**Comment by norg** [comment]
Doc looks good to me

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -590,6 +592,70 @@ Notes
 
 -  further notes at the ``file_data`` section below.
 
+http.server
```
Comment (victorjulien) [inline doc/userguide/rules/http-keywords.rst]: this is a copy of the file_data or http server body, and that makes it almost completely wrong. It's better to look at some of the other keywords registered using the same 'stub', for example http_accept and http_accept_enc.

Comment (jlucovsky) [reply doc/userguide/rules/http-keywords.rst]: Updated documentation for `http.server` and `http.location`

**Comment by jlucovsky** [comment]
Continued in #3712

**Pull request closed without merging**